### DISCUSSION TITLE
コメント入力欄付近でのユーザーごとのコメント募集状況の設定の表示

### DISCUSSION THREAD

**Discussion by simon-ritchie** [increments/qiita-discussions#115] [要望・アイデア・不具合]
## 前置き

※ぼんやりとこんな感じの機能があったらより快適になりそうだなぁ・・・といったくらいの何となく考えているくらいの要望です！特に急ぎで必要・・・とかではないので、優先度低めで運営の皆様方の頭の片隅に残ったら良いな・・・というものになります！

## 欲しい機能

- ユーザーごとに設定ができる形で、コメントをどんな感じに募集しているのかを設定できたら良いなと思います！
- 設定した内容はコメント入力欄の付近に目立つように表示される形を想定しております！
- 後述しますが、特に新規ユーザーに取ってもコミュニティで快適に過ごせるか・・・に絡んでくるので新規登録時に3～4択などで選択して初期設定する形だとより良いかもしれません！（「全面的に受け付ける」「やさしめの内容のみ受け付ける」「受け付けない」などなど）

## なぜこの機能が欲しいと思ったか

コメント機能自体は知識の漏れなどをご指摘いただいて勉強になったりなどとても有益に感じております。

一方で他人に対するコメントを眺めていたりしていてたまに「このコメントは結構雰囲気が厳しく感じられる・・・」といったケースもちらほら見かけることがあります。

コメントなどをされている方は善意でされている方がほとんどだと思いますし、雰囲気を厳しくするという意図はない方が大半だとは思っております。この辺はQiita上だとどうしてもテキストコミュニケーションになるので（記号や絵文字などを使ったりなどして）気を付けていないとどうしても厳しい雰囲気に感じられてしまったりといったケースが発生しうると感じております。

例 : 少し前に公開されたサイボウズさんの記事より引用:

> 熱田：これを言ったメンバーが、特別性格が悪いとかではぜんぜんないんですよね。普通に対面で話すとめちゃめちゃいい人なんですが（笑）。サイボウズあるあるですよね。
...
> 熱田：対面で話すと優しいんだけれども、テキストだとちょっと厳しい言葉を受けることがある。

[在宅勤務はもう限界……テレワークつらい勢の悲痛な叫び　リモート9割のサイボウズで、社員が「つらみ」を感じるポイント](https://logmi.jp/business/articles/325665)

私とかは豆腐メンタルなのでマサカリは弱めだと助かり、且つコメントのご指摘は勉強になるのでなるべく受け付けたい・・・という感じですが、この辺はどこまで許容できるかはユーザーによりけりだと思います（例 : 全然余裕で色々なコメントがウェルカムな方、優し目なコメントであれば大歓迎な方、コメントは受け付けたくない方etc）。

特にテックブログやコミュニティなどに不慣れな新規ユーザーの方とかが厳しめなことを言われてしまうと「なんだかQiitaで記事書くの怖い・・・」といった風に感じてしまって執筆などを控えたりしてしまう方もいらっしゃるかもしれません。この辺は（計測しにくいところではありますが）Qiitaに取っても不利益になりうると思います。

※この辺ははてブであればコメントを受け付けないようにしたりなるべくポジティブなものを目立たせるようにしたり、Twitterであればフォローなどしている人のみコメントに参加できる設定があるとか、SNSやコミュニティによって様々ですが・・・。

そのため例えば優し目なコメントのみ歓迎しているようなユーザーであればコメント入力欄の付近に「※このユーザーは優し目なコメントのみを歓迎しています」的なメッセージが目立つように表示できたり・・・するとユーザーに取っても精神衛生上的に好ましく、且つコミュニティ上で委縮せず気軽な活動をしやすくなる・・・と感じております！

※コミュニティガイドラインにもHRTのところなどで触れられていますが、テキストコミュニケーションという性質上ガイドラインを読んだ方々でも雰囲気が怖くなりがちなケースは発生しうると思いますので、より一層コミュニティの雰囲気を良いものにするためにこういった機能が欲しいと思っております！（コメント欄などでの喧嘩なども減った方が運営の皆さんの負担も減るのでは・・・という気もしております）

**Comment by simon-ritchie** [comment]
> 特にテックブログやコミュニティなどに不慣れな新規ユーザーの方とかが厳しめなことを言われてしまうと「なんだかQiitaで記事書くの怖い・・・」といった風に感じてしまって執筆などを控えたりしてしまう方もいらっしゃるかもしれません。この辺は（計測しにくいところではありますが）Qiitaに取っても不利益になりうると思います。

余談となりますが、計測はしにくいとは書きましたが優し目のコメントのみ受け付けるユーザーに対して「ポジティブなコメントが多くなっているかどうか」とかの辺りはAWSのComprehendとか使えばデータサイエンティストさんみたいな詳しい方抜きに感情値など定量的に計測できるのでその辺で計測してみて、効果が認められなければ将来廃止する・・・とかでも良いのかもしれません！（もし可能であればまずは試しで運用してみて、効果が無いのに設定で残っているとUIが複雑になるので計測して残すか判断するなど）
（たくさんのデータに対して数値化する・・・とかだとAWS SDKとかの操作は必要にはなりますが・・・）

例 : [テキストの感情を分析する](https://aws.amazon.com/jp/getting-started/hands-on/analyze-sentiment-comprehend/)

日々の全コメントデータを対象とすると膨大になってクラウドコストも高くつく・・・とは思いますが、単発の分析＆対象データを絞って実施・・・とかなら大したコストにならずに対応できるとは思います！

※Qiitaのコメントでもうまいこと判定してくれるかどうかは試しておりませんが、以前仕事で触っていた時には日本語でも結構精度が良くなってきたように感じております。

※この辺は触っていませんがGCPとかでも似たようなものがある・・・と思います。

**Comment by kskwtnk** [comment]
@simon-ritchie 
具体的なソリューションまで含めてのご提案ありがとうございます！

この件、まさに運営としては悩んでいることでして非常に嬉しいです！
実際に提供できるのか、どのような機能になるのか……は正直言って全然見えていないものの、方向性としては是非とも実施したい内容です

Qiitaの思想としてもHRTを大事にしていきたいですし、私自身もいただいたコメントで怖い思いをしたことはあります
繰り返しになりますが今回こちらのご意見をいただいて非常に嬉しいです！参考にさせていただきます！
